src/core/EventBus.js

```diff
@@ -49,11 +51,13 @@ function EventBus() {
 
         var handler = {
             callback: listener,
-            scope: scope
+            scope: scope,
+            priority: priority !== undefined ? priority : LOW_PRIORITY
         };
 
         handlers[type] = handlers[type] || [];
         handlers[type].push(handler);
+        handlers[type].sort(function (a, b) {return a.priority - b.priority;});
```

Comment:
It feels like it would be better not to have to worry about the order in which the handlers are registered. 
If there's a dependency on the order the handlers are run, it would be better to refactor?
It also seems error prone if someone was to add multiple handlers with the same (non-LOW_PRIORITY) priority since they would end up being inserted in registration order.